lib/Controller/ContactIntegrationController.php

```diff
@@ -75,8 +80,13 @@ public function newContact(?string $contactName = null, ?string $mail = null): J
 	 * @return JSONResponse
 	 */
 	#[TrapError]
-	public function autoComplete(string $term): JSONResponse {
-		$res = $this->service->autoComplete($term);
-		return (new JSONResponse($res))->cacheFor(60 * 60, false, true);
+	public function autoComplete(string $term, bool $forceSAB = false): JSONResponse {
+		$cached = $this->cache->get($term);
+		if($cached !== null) {
+			return new JSONResponse(json_decode($cached));
+		}
+		$res = $this->service->autoComplete($term, $forceSAB);
+		$this->cache->set($term, json_encode($res), 7 * 24 * 3600);
```

Comment:
1 week is a bit long. Do an hour, that will catch frequently mentioned people and still let changes propagate